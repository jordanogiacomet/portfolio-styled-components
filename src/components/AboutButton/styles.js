import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
