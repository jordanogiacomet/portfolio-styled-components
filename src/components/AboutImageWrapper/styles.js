import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
