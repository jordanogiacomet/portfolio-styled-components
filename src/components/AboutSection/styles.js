import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
