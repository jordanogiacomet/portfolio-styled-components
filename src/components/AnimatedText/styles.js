import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;