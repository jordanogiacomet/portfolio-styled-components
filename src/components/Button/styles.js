import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 