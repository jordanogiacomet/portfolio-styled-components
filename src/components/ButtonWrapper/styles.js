import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;