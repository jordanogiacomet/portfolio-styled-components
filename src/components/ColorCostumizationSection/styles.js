// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
