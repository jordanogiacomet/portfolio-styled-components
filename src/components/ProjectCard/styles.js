import styled from 'styled-components';

export const Container = styled.div`
  width: 300px; 
  height: 300px; 
  background: ${({ theme }) => theme.colors.heroWrapperBackground}
  border-radius: 0.5rem;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: relative;
  transition: transform 0.2s ease-in-out;
  @media ${({ theme }) => theme.media.md} {
    &:hover {
    transform: scale(1.05);
    };
  };
  @media ${({ theme }) => theme.mediaMax.md} {
    width: 300px;
  };
`;

export const Heading = styled.h2`
  position: absolute;
  z-index: 2;
  transition: opacity 0.2s ease-in-out, visibility 0.2s ease-in-out;
  color: ${({ theme }) => theme.colors.white};
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;;
  top: 45%;
  width: 100%;
  line-break: auto;
  font-size: ${({ theme }) => theme.font.sizes.medium};
 
  @media ${({ theme }) => theme.mediaMax.md} {
    width: 100%;
    top: 75%;
  };
  @media ${({ theme }) => theme.mediaMax.sm} {
    font-size: ${({ theme }) => theme.font.sizes.lg};
  };
  @media ${({ theme }) => theme.media.md} {
    bottom: 1rem;
    &.hide {
    opacity: 0;
    visibility: hidden;
    };

  &.show {
    opacity: 1;
    visibility: visible;
    };
  };
`;
