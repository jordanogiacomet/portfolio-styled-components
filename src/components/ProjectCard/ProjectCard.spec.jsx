import { renderTheme } from '../../styles/render-theme';
import { ProjectCard } from '.';

describe('<ProjectCard />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <ProjectCard
        imgUrl="/test.jpg"
        title="Test Project"
        gitUrl="/git"
        previewUrl="/preview"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render ProjectCard with all props', () => {
    const { getByText } = renderTheme(
      <ProjectCard
        imgUrl="/test.jpg"
        title="Test Project"
        gitUrl="/git"
        previewUrl="/preview"
      />
    );
    expect(getByText('Test Project')).toBeInTheDocument();
  });
});