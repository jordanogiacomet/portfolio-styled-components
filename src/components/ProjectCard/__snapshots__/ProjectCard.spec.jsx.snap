// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectCard /> should match snapshot 1`] = `
.c0 {
  width: 300px;
  height: 300px;
  background: #181818;
  border-radius: 0.5rem;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: relative;
  transition: transform 0.3s ease-in-out;
}

.c0:hover {
  transform: scale(1.05);
}

.c4 {
  position: absolute;
  z-index: 2;
  transition: opacity 0.3s ease-in-out,visibility 0.3s ease-in-out;
  color: white;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
  top: 50%;
  width: 100%;
  line-break: auto;
  font-size: 2.5rem;
}

.c4.hide {
  opacity: 0;
  visibility: hidden;
}

.c4.show {
  opacity: 1;
  visibility: visible;
}

.c1 {
  border-top-left-radius: 1rem;
  border-top-right-radius: 1rem;
  background: url(/test.jpg) no-repeat center center/cover;
  position: relative;
  height: 100%;
}

.c3 {
  height: 3.5rem;
  width: 3.5rem;
  border: 0.125rem solid #adb7be;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  transition: border-color 0.3s ease-in-out,visibility 0.2s ease-in-out,opacity 0.2s ease-in-out;
  visibility: hidden;
  opacity: 0;
}

.c3:hover {
  border-color: white;
}

.c3:hover .icon {
  color: white;
}

.c3 .icon {
  height: 2.5rem;
  width: 2.5rem;
  color: #adb7be;
  transition: color 0.3s ease-in-out;
}

.c2 {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(24, 24, 24, 0.8);
  z-index: 1;
}

.c2.show {
  opacity: 1;
  visibility: visible;
}

.c2:hover a {
  visibility: visible;
  opacity: 1;
  transition: opacity 0.2s ease-in-out,visibility 0.2s ease-in-out;
}

@media (max-width: 768px) {
  .c0 {
    width: 100%;
  }
}

@media (max-width: 768px) {
  .c4 {
    width: 100%;
  }
}

@media (max-width: 640px) {
  .c4 {
    font-size: 1.8rem;
  }
}

@media (min-width: 768px) {
  .c4 {
    bottom: 1rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      imgurl="/test.jpg"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/git"
        >
          <svg
            aria-hidden="true"
            class="icon"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.25 6.75 22.5 12l-5.25 5.25m-10.5 0L1.5 12l5.25-5.25m7.5-3-4.5 16.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
        <a
          class="c3"
          href="/preview"
        >
          <svg
            aria-hidden="true"
            class="icon"
            data-slot="icon"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </a>
      </div>
      <h2
        class="c4 show"
      >
        Test Project
      </h2>
    </div>
  </div>
</div>
`;
